packages/ui-heading/src/Heading/theme.js

```diff
@@ -22,58 +22,80 @@
  * SOFTWARE.
  */
 
-export default function generator({ borders, colors, spacing, typography }) {
-  return {
-    lineHeight: typography.lineHeightFit,
+/**
+ * Generates the theme object for the component from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @return {Object} The final theme object with the overrides and component variables
+ */
+const generateComponentTheme = (theme, themeOverride = {}) => {
+  const {
+    typography,
+    brandVariables,
+    colors,
+    spacing,
+    borders,
+    key: themeName
+  } = theme
 
-    h1FontSize: typography.fontSizeXXLarge,
-    h1FontWeight: typography.fontWeightLight,
-    h1FontFamily: typography.fontFamily,
+  const themeSpecificStyle = {
+    canvas: {
+      primaryColor: brandVariables
+        ? brandVariables['ic-brand-font-color-dark']
+        : ''
```

Comment:
Just another thing caught my eye: I remember you made a change on v8 branch about brandVariables, but I don't think it got cherrypicked here, so I'm not sure this would work. In the other components we use `theme['ic-brand-font-color-dark']`. (Btw, I would probably leave it like on the theme object: not sure if brandVariables would break anything for our users.)

Reply:
Good catch, thanks a lot